### ISSUE TITLE
[Issue]: Image artifacts after upscaling

### ISSUE THREAD

**Issue by Threnos** [vladmandic/sd-extension-chainner#6]
### Issue Description

Sorry if this question was asked before (I tried to look for one as much as I could), but have anyone experienced artifacts in upscaled images?
They seem to appear and dissapear randomly, varying in strength from just couple barely noticeble dots to what is shown on example, even on ESRGAN upscalers that pre-update always worked just fine.

For example chaiNNer 4xNomos8kSCHAT-L output:
![image](https://github.com/vladmandic/automatic/assets/30055497/493d94f0-c438-451a-a38b-5439d11c703f)

### Version Platform Description

Version: app=sd.next updated=2023-10-18 hash=66f75b6a url=https://github.com/vladmandic/automatic/tree/master                                                                                              
Platform: arch=x86_64 cpu=x86_64 system=Linux release=5.15.0-86-generic python=3.10.12                                                                                                                     
nVidia CUDA toolkit detected: nvidia-smi present
Engine: backend=Backend.ORIGINAL compute=cuda mode=no_grad device=cuda cross-optimization="Scaled-Dot-Product"
Device: device=NVIDIA GeForce RTX 3050 Ti Laptop GPU n=1 arch=sm_90 cap=(8, 6) cuda=12.1 cudnn=8902 driver=535.113.01
Torch override VAE dtype: no-half set
Setting Torch parameters: device=cuda dtype=torch.float32 vae=torch.float32 unet=torch.float32 context=inference_mode fp16=False bf16=False
--medvram

### Backend

Original

### Model

SD 1.5

### Acknowledgements

- [X] I have read the above and searched for existing issues
- [X] I confirm that this is classified correctly and its not an extension issue

**Comment by xGeekpower** [comment]
maybe the nvdia driver is messing things up?
https://github.com/vladmandic/automatic/discussions/1285
https://www.nvidia.com/download/driverResults.aspx/204771/en-us/

Also if you did a clean install for the latest update you might wanna check if all the settings you had are the same (this happened to me didn't notice that by default some things are now checked that I didn't used before and made everything bad)

**Comment by Threnos** [comment]
@xGeekpower
On linux we don't have such luxury as VRAM to RAM offloading.

The installation was clean, just as recommended.

Definitely not all settings are the same as they used to be before update, but upscaler-related ones are. But I never ever experienced such issue before, no matter what experiments were done with them.

**Comment by mart-hill** [comment]
I know you don't like such statements, @vladmandic, but I've noticed that weird neon-ing on Windows 10 22H2/RTX 3090/537.42 Studio driver, too, with Chainner scalers, that work (the ones, which don't give black image). This also might 'pollute' later upscalings with ERSGAN scalers, I had to either generate a bit more images with changing scalers in-between, or just restart the UI completely.
I hope this observation helps a bit. 🙂

**Comment by vladmandic** [comment]
@mart-hill it does and i was aware of chainner models sometimes clipping.
unfortunately, this issue was opened with comment "even on ESRGAN upscalers that pre-update always worked just fine" which didn't make sense.

anyhow, i've just updated chainner extension, should be fixed.

**Comment by Threnos** [comment]
I saw a commit that said domething about normalization of upscaled image. It did not help with my issue.

**Comment by vladmandic** [comment]
what issue? exact upscaler and exact artifacts. does it happen with all upscalers or just some. etc.

**Comment by MaxJa4** [comment]
Running in the same issue as the issue author.

Here is a sample XYZ grid with all chainner upscalers.
Using diffusers backend with SD1.5, commit bc80c14 of SD.Next and commit 9d5e6c2 of the chainner extension.
<details><summary>XYZ grid (click to open)</summary>

![tmp6h91_l2a](https://github.com/vladmandic/automatic/assets/74194322/40d531d2-2ac3-47c2-a56e-0d67d6518dc5)

</details> 

TLDR: Everyone but SwiftSR ones are affected. Not SD model dependent. No issues with the "normal" ones like UltraSharp, NMKD's upscalers, R-ESRGAN, etc. (used one NMKD upscaler at the end for reference, no issues there).

**Comment by vladmandic** [comment]
> Running in the same issue as the issue author.

Post EXAMPLE and exact version of chainner extension.

**Comment by vladmandic** [comment]
i thought that would not happen when i made the switch from numpy decode to torchvision, but looks like re-norm is still required. should be fixed now.

**Comment by MaxJa4** [comment]
Just tested it, I feel like it's significantly better now in some cases (could be the seed though; now using sd-chainner 2488b23).
It's still clipping in some areas though, see the xyz grid below. Best visible on the left most chair.

![tmpwoj7cspl](https://github.com/vladmandic/automatic/assets/74194322/062317a2-300f-4923-acbe-f7b3b3fc0ac1)

**Comment by MaxJa4** [comment]
Issue found and fixed --> PR created

**Comment by vladmandic** [comment]
reopening and transferring to correct repo.

**Comment by vladmandic** [comment]
all here - just did a push, can you update and give it a quick test? i'll keep open this time until confirmed as resolved.

**Comment by MaxJa4** [comment]
![tmpz0idi3n0](https://github.com/vladmandic/sd-extension-chainner/assets/74194322/f992b30c-48b8-4151-b43b-e542e6fb75f4)

Did some testing with bright and dark images, where it would clip previously.
All chainner upscalers look correct regarding brightness and color in comparison to "normal" upscalers like UltraSharp or NMKD-Superscale (used here for reference).

Looks fixed for me! Thanks.
